packages/widgets/src/HeaderBar/Apps.js

```diff
@@ -84,194 +83,218 @@ Search.propTypes = {
+    <div data-test="headerbar-apps-menu">
+        <Card>
+            <Search value={filter} onChange={onFilterChange} />
+            <List apps={apps} filter={filter} />
+        </Card>
+
+        <style jsx>{`
+            div {
+                z-index: 10000;
+                position: absolute;
+                top: 28px;
+                right: -6px;
+                border-top: 4px solid transparent;
+            }
+        `}</style>
+    </div>
+)
+
+AppMenu.propTypes = {
+    apps: propTypes.array.isRequired,
+    onFilterChange: propTypes.func.isRequired,
+    filter: propTypes.string,
+}
 
-    componentWillUnmount() {
-        document.removeEventListener('click', this.onDocClick)
-    }
+const mutation = {
+    resource: 'i18n',
+    type: 'create',
+    data: ({ items }) => items,
+}
 
-    onDocClick = evt => {
-        if (this.elContainer && !this.elContainer.contains(evt.target)) {
-            this.setState({ show: false })
+const Apps = ({ apps }) => {
+    const [translations, setTranslations] = useState({})
+    const [show, setShow] = useState(false)
+    const [filter, setFilter] = useState('')
+
+    const handleVisibilityToggle = useCallback(() => setShow(!show), [show])
+    const handleFilterChange = useCallback(({ value }) => setFilter(value), [])
+
+    const [mutate] = useDataMutation(mutation, {
+        onComplete: translations => setTranslations(translations),
+    })
+    useEffect(() => {
+        mutate({
+            items: apps.map(({ name }) => name),
+        })
+    }, [apps])
+
+    const containerEl = useRef(null)
+    const onDocClick = useCallback(evt => {
+        if (containerEl.current && !containerEl.current.contains(evt.target)) {
+            setShow(false)
         }
-    }
-
-    onToggle = () => this.setState({ show: !this.state.show })
+    }, [])
+    useEffect(() => {
+        document.addEventListener('click', onDocClick)
+        return () => document.removeEventListener('click', onDocClick)
+    }, [])
 
-    onChange = ({ value }) => this.setState({ filter: value })
-
-    AppMenu = apps => (
-        <div data-test="headerbar-apps-menu">
-            <Card>
-                <Search value={this.state.filter} onChange={this.onChange} />
-                <List apps={apps} filter={this.state.filter} />
-            </Card>
+    return (
+        <div ref={containerEl} data-test="headerbar-apps">
+            <button
+                onClick={handleVisibilityToggle}
+                data-test="headerbar-apps-icon"
+            >
+                <AppsIcon className={appIcon.className} />
+            </button>
+
+            {show ? (
+                <AppMenu
+                    apps={apps.map(app => ({
+                        ...app,
+                        displayName: translations[app.name] || app.displayName,
+                    }))}
+                    filter={filter}
+                    onFilterChange={handleFilterChange}
+                />
+            ) : null}
```

Comment:
We should probably show a loading state here or within <AppMenu>... this will currently flash the English name before the translations load, which isn't what we want.